hphp/runtime/vm/jit/vasm-arm.cpp

```diff
@@ -117,6 +117,47 @@ vixl::Condition C(ConditionCode cc) {
   return arm::convertCC(cc);
 }
 
+/*
+ * Uses the flags from the Vinstr which defs SF to determine
+ * whether or not the Vixl assembler should emit code which
+ * sets the status flags.
+ */
+vixl::FlagsUpdate updateFlags(uint8_t flags) {
+  return flags ? SetFlags : LeaveFlags;
+}
+
+#define UF(x) updateFlags(x)
+
+/*
+ * There are numerous ARM instructions that don't set status flags, and
+ * therefore those flags must be set synthetically in the emitters. This
+ * assertion is applied to the emitters which don't set all of the status
+ * flags required by the Vinstr which defs SF. The flags field of the
+ * Vinstr is used to determine which bits are required. Those required
+ * bits are compared against the bits which are actually set by the
+ * implementation.
+ */
+template<class Inst> void checkSF(const Inst& i, StatusFlags s) {
+  uint8_t required = i.fl;
+  uint8_t set = static_cast<uint8_t>(s);
+  always_assert_flog((required & set) == required,
+                     "should def SF but does not: {}\n",
+                     vinst_names[Vinstr(i).op]);
+}
+
+template<class Inst> void checkSF(const Inst& i) {
+  checkSF(i, StatusFlags::None);
+}
```

Comment:
Originally, I used this function to protect instructions which didn't support setting status flags. With all of the SF setting tidied up, we can probably remove it.